src/sql/migrations/20230907104002_cell_metadata_tables.js

```diff
@@ -0,0 +1,51 @@
+const defineDeleteCellMetadataFileIfOrphanFunc = `
+  CREATE OR REPLACE FUNCTION delete_cell_metadata_file_if_orphan()
+    RETURNS trigger AS $$
+    BEGIN
+      DELETE FROM cell_metadata_file
+      WHERE
+        cell_metadata_file.id = OLD.cell_metadata_file_id AND
+        NOT EXISTS (
+          SELECT FROM cell_metadata_file_to_experiment cm_map
+          WHERE cm_map.cell_metadata_file_id = OLD.cell_metadata_file_id
+        );
+      RETURN OLD;
+    END;
+    $$ LANGUAGE plpgsql;
+`;
+
+const createDeleteCellMetadataFileIfOrphanTrigger = `
+  CREATE TRIGGER delete_cell_metadata_file_if_orphan_trigger
+  AFTER DELETE ON cell_metadata_file_to_experiment
+  FOR EACH ROW
+  EXECUTE FUNCTION delete_cell_metadata_file_if_orphan();
+`;
+
+exports.up = async (knex) => {
+  await knex.schema.createTable('cell_metadata_file', (table) => {
+    table.uuid('id').primary();
+    table.string('name', 255).notNullable();
+    table.enum('upload_status', ['PENDING', 'IN_PROGRESS', 'COMPLETED', 'FAILED']).defaultTo('PENDING');
```

Comment:
Does this work like with the native enum? Or does it create a new one?

Should we perhaps reuse the existing one?